foreshadow/core/serializers.py

```diff
@@ -0,0 +1,463 @@
+        else:
+            return {}
+
+    def dict_serialize(self, deep=True):
+        """Serialize the init parameters (dictionary form) of a transformer.
+
+        Args:
+            deep (bool): If True, will return the parameters for this estimator
+                recursively
+
+        Returns:
+            dict: The initialization parameters of the transformer.
+
+        """
+        return {
+            "data": _make_serializable(
+                self.get_params(deep), serialize_args=self.serialize_params
+            )
+        }
+
+    @classmethod
+    def dict_deserialize(cls, data):
+        """Deserialize the dictionary form of a transformer.
+
+        Args:
+            data: The dictionary to parse as the transformer is constructed.
+
+        Returns:
+            object: A re-constructed transformer
+
+        """
+        params = _make_deserializable(data["data"])
+        pickle_class = data.get("pickle_class")
+        if pickle_class is not None:
+            pickle_class = _unpickle_inline_repr(pickle_class)
+            return pickle_class(**params)
+        else:
+            # Cannot use set_params since steps is a required init arg
+            # for Pipelines
+            return cls(**params)
+
+    def inline_serialize(self):
+        """Convert transformer to hex pickle form inline in a dictionary form.
+
+        Returns:
+            A string representation of the pickle dump
+
+        """
+        return {"data": _pickle_inline_repr(self)}
+
+    @classmethod
+    def inline_deserialize(cls, data):
+        """Unpickle an inline pickled transformer.
+
+        Args:
+            data: The dictionary data of the transformer.
+
+        Returns:
+            object: The constructed transformer.
+
+        """
+        return _unpickle_inline_repr(data["data"])
+
+    def disk_serialize(self, cache_path=None):
+        """Convert transformer to pickle and save it disk in a cache directory.
+
+        Args:
+            cache_path (str): Override the default cache path which is in the
+                root of the user directory
+
+        Returns:
+            str: The path the data was saved to.
+
+        """
+        fpath = _pickle_cache_path(cache_path)
+        with open(fpath, "wb+") as fopen:
+            pickle.dump(self, fopen, protocol=pickle.HIGHEST_PROTOCOL)
+
+        return {"data": fpath}
+
+    @classmethod
+    def disk_deserialize(cls, data):
+        """Deserialize a transformer disk cache serialized.
+
+        Args:
+            data: The dictionary data of the transformer.
+
+        Returns:
+            object: The constructed transformer.
+
+        """
+        fpath = data["data"]
+        with open(fpath, "rb") as fopen:
+            return pickle.load(fopen)
+
+    def serialize(self, method=None, name=None, **kwargs):
+        """Serialize data as specified.
+
+        If you would like to save the transformer parameters without saving
```

Comment:
'in the serialized form'

Reply:
Actually on second look, I think it might make less sense if I change this.